pkgs/build-support/php/build-composer-project.nix

```diff
@@ -62,24 +64,25 @@ let
         runHook postInstallCheck
       '';
 
-      composerRepository = phpDrv.mkComposerRepository {
+      composerRepository = finalAttrs.php.mkComposerRepository {
         inherit composer composer-local-repo-plugin;
         inherit (finalAttrs) patches pname src vendorHash version;
 
+        php = finalAttrs.php;
         composerLock = previousAttrs.composerLock or null;
         composerNoDev = previousAttrs.composerNoDev or true;
         composerNoPlugins = previousAttrs.composerNoPlugins or true;
         composerNoScripts = previousAttrs.composerNoScripts or true;
         composerStrictValidation = previousAttrs.composerStrictValidation or true;
       };
 
-      COMPOSER_CACHE_DIR="/dev/null";
-      COMPOSER_DISABLE_NETWORK="1";
-      COMPOSER_MIRROR_PATH_REPOS="1";
+      COMPOSER_CACHE_DIR = "/dev/null";
```

Comment:
Also unrelated but this should go to `env`: https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/environment-variables-go-to-env.md

Reply:
Fixed too, thanks for this, I've learned something new.